src/lib.rs

```diff
@@ -413,6 +416,126 @@ impl Contract {
             .collect()
     }
 
+    pub fn get_addon(&self, id: CommunityAddOnId) -> Option<CommunityAddOn> {
+        self.available_addons.get(&id)
+    }
+
+    pub fn get_available_addons(&self) -> Vec<CommunityAddOn> {
+        self.available_addons.iter().map(|(_id, add_on)| add_on).collect()
+    }
+
+    // Only the contract admin and DevHub moderators
+    pub fn create_new_addon(&mut self, addon: CommunityAddOn) {
+        if !self.has_moderator(env::predecessor_account_id())
+            && env::predecessor_account_id() != env::current_account_id()
+        {
+            panic!("Only the admin and moderators can create new add-ons");
+        }
+        if self.get_addon(addon.id.to_owned()).is_some() {
+            panic!("Add-on with this id already exists");
+        }
+        addon.validate();
+        self.available_addons.insert(&addon.id.clone(), &addon);
+    }
+
+    // ONLY FOR TESTING
+    pub fn delete_addon(&mut self, id: CommunityAddOnId) {
+        // Also delete from communities
+        if !self.has_moderator(env::predecessor_account_id())
+            && env::predecessor_account_id() != env::current_account_id()
+        {
+            panic!("Only the admin and moderators can delete add-ons");
+        }
+        let addon = self
+            .get_addon(id.clone())
+            .expect(&format!("Add-on with id `{}` does not exist", id))
+            .clone();
+
+        // let mut new_communities: UnorderedMap<CommunityHandle, Community> =
+        //     UnorderedMap::new(StorageKey::Communities);
+
+        // for (community_handle, mut community) in self.communities.iter() {
+        //     // Try to remove add on from community
+        //     community.remove_addon(addon.id.to_owned());
+        //     new_communities.insert(&community_handle, &community);
+        // }
+        // self.communities = new_communities;
+        self.available_addons.remove(&addon.id);
+    }
+
+    pub fn edit_addon(&mut self, input: CommunityAddOn) {
+        if !self.has_moderator(env::predecessor_account_id())
+            && env::predecessor_account_id() != env::current_account_id()
+        {
+            panic!("Only the admin and moderators can edit add-ons");
+        }
+        self.available_addons.insert(&input.id.clone(), &input);
+    }
+
+    pub fn get_community_addons(&self, handle: CommunityHandle) -> Vec<CommunityAddOn> {
+        let community = self
+            .get_community(handle.to_owned())
+            .expect(format!("Community with handle `{}` does not exist", handle).as_str());
+        community
+            .addon_list
+            .iter()
+            .map(|addon_config| {
+                self.get_addon(addon_config.addon_id.clone()).expect("add-on missing")
+            })
+            .collect()
+    }
+
+    pub fn add_community_addon(
```

Comment:
Could we have this method generate the addon config id?

See this transaction panic because config_id is not set: https://explorer.testnet.near.org/transactions/8THKLRov1FAKbj7iSu4vhCX3NniBzqvsh3a221PqZqXH

Could be considered with Peter's suggestion

Reply:
I think it would be better to generate or choose such an id from the front end side.

Reply:
We won't need Config ID in the first place, if we just move the list of Configs to the Community struct.

Reply:
> would it be better to use a `LookupMap` for the community plugin configuration data? That way it would be possible to look up the config for the plugin in view, and also not make the community config object potentially very large.

@frol, @petersalomonsen adviced that this would cause the community to become too big in some cases.

Reply:
Just to bring more context from the thread below: https://github.com/near/neardevhub-contract/pull/61#discussion_r1360615452

TL;DR: I don't think it would be too big, and most probably this would use less gas, but I vote for moving fast here and don't hold it back any longer. This already took too much time, really.